util/pin/types/pintypes.go

```diff
@@ -0,0 +1,17 @@
+// Package pintypes provides pin types
+package pintypes
+
+import binfotypes "github.com/moby/buildkit/util/buildinfo/types"
+
+// Source corresponds to buildinfo Source.
+type Source = binfotypes.Source
+
+// Pin provides pinning information to ensure determinism of sources.
+type Pin struct {
```

Comment:
@tonistiigi 
Should this have a `version` field or `mediaType` field for extensibility?
https://github.com/moby/buildkit/issues/2794#issuecomment-1094895736